pandas/tests/test_categorical.py

```diff
@@ -3625,6 +3625,39 @@ def test_cat_accessor_api(self):
             invalid.cat
         self.assertFalse(hasattr(invalid, 'cat'))
 
+    def test_str_accessor_api_for_categorical(self):
+        # https://github.com/pydata/pandas/issues/10661
+        from pandas.core.strings import StringMethods
+        s = Series(list('aabb')).astype('category')
+        self.assertIsInstance(s.str, StringMethods)
+
+        res = s.str.contains("a")
```

Comment:
we need tests for all the methods as I suspect some don't work (or do they all?)